apix-rest-v1/src/main/java/eu/xenit/apix/rest/v2/ApixV2Webscript.java

```diff
@@ -44,6 +44,17 @@
             boolean retrieveTargetAssocs, boolean retrieveSourceAssocs) {
         List<NodeInfo> nodeInfoList = new ArrayList<NodeInfo>();
         for (NodeRef nodeRef : nodeRefs) {
+
+            if (!permissionService.hasPermission(nodeRef, IPermissionService.READ)) {
+                logger.warn("Excluding node {} from results due to insufficient permissions", nodeRef);
+                continue;
+            }
+
+            if (!nodeService.exists(nodeRef)) {
```

Comment:
Maybe change the order of these checks. NodeService.exists() is the shorter operation

Reply:
In effect nothing changes by changing the order in this particular case, but the exists check will apparently always return true if a user has insufficient permissions to check for that node (dixit axel faust). Given this I'd keep it as is.